src/Microsoft.DotNet.Build.Tasks/GenerateVersionInfoFile.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateVersionInfoFile : Task
+    {
+        [Required]
+        public ITaskItem InfoFile { get; set; }
+
+        [Required]
+        public string LatestCommit { get; set; }
+
+        [Required]
+        public string VersionSeedDate { get; set; }
+
+        [Required]
+        public string VersionSeedSourceComment { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                string path = InfoFile.GetMetadata("FullPath");
+                string directory = Path.GetDirectoryName(path);
+                if (!Directory.Exists(directory))
+                {
+                    Directory.CreateDirectory(directory);
+                }
+                File.WriteAllLines(path, new []
+                {
+                    @"<?xml version=""1.0"" encoding=""utf-8""?>",
```

Comment:
Any reason you need a custom task to do this? You can just call WriteLinesToFile task like the other props file https://github.com/dotnet/buildtools/pull/1191/files#diff-09dfec4a89fd90309644b01711f96159R307.